### DISCUSSION TITLE
[Question]: It logout when close webpage.

### DISCUSSION THREAD

**Discussion by zhaolihang** [danny-avila/LibreChat#4239] [Troubleshooting] [answered]
### What is your question?

It logout when close webpage. or press F5 refresh page.

### More Details

 It start by :
 `docker compose -f ./deploy-compose.yml up -d`
ubuntu22.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxesse** [comment, marked as answer]
I guess you're accessing it by IP? It will do that. Try to use localhost, or a DNS name and it won't log you out.

**Reply by zhaolihang** [reply]
thanks

**Reply by Filipe-Araujo0** [reply]
why?

**Comment by GChristensen** [comment]
This should be a pinned thread or highlighted by two blinking borders in the installation guide. I installed it in the docker (which is in WSL under Windows) and got the same issue accessing from LAN. It turns out, that the problem is the unsecure protocol (HTTP vs HTTPS), under which cookies aren't sent. The solution is here: 

https://github.com/danny-avila/LibreChat/discussions/572

I've got it working after adding the following into docker-compose.override.yml 

```
services:
  api:
     command: npm run backend:dev
     environment: 
       - NODE_ENV=development
```

And the following into .env

`NODE_ENV=development`

Although I don't actually know if the latter is needed.
It is necessary to run `docker compose build` and `docker compose restart` after that.

**Comment by kamaldua65** [comment]
https://www.librechat.ai/blog/2024-05-16_unsecured_http
